### DISCUSSION TITLE
Re-source `.obsidian.vimrc`

### DISCUSSION THREAD

**Discussion by huyz** [esm7/obsidian-vimrc-support#153] [Q&A]
Is there a way to re-source `.obsidian.vimrc` without restarting Obsidian?

**Comment by connerohnesorge** [comment]
You can reload obsidian using the command to reload the app without saving through command pallette or  you could setup a hotkey for it.

**Reply by huyz** [reply]
That's slightly faster than quitting out of Obsidian and re-launching it.  So thanks.

But I was still hoping for reloading just this plugin.

**Reply by connerohnesorge** [reply]
You could also reload just the plugins in the community plugins tab

**Reply by huyz** [reply]
I'll try to write a JS script to disable and re-enable the plugin.

**Reply by connerohnesorge** [reply]
This would be awesome as part of the plugin!

**Reply by esm7** [reply]
Note that while the basic action of re-sourcing `.obsidian.vimrc` is very simple, either by baking it into the plugin code or a JS that forces it to reload, I'm concerned about side-effects. Anyone testing a solution in one of these directions should pay special care to make sure things like registering events, CodeMirror hooks etc don't get duplicated on every reload.

**Reply by huyz** [reply]
Thanks for letting us know @esm7.  I'm not sure when I'll be able to get to it, but I'll keep that in mind if and when I do.

**Reply by huyz** [reply]
Relevant plugin mentioned here: https://forum.obsidian.md/t/hot-reload-plugins-when-changes-detected-on-file/49467

**Reply by adamreiswig** [reply]
> Relevant plugin mentioned here: https://forum.obsidian.md/t/hot-reload-plugins-when-changes-detected-on-file/49467

Would you be able to put his up on github? My work block forum.obsidian.md. :(

**Reply by huyz** [reply]
https://github.com/pjeby/hot-reload

**Comment by artemDvoryadkin** [comment]
hotreload, of course it works, but in open files in visual mode it doesn't work correctly, my approach is simpler.
the sequence is as follows
yy : enter
that is, I copy the line that changed in the vimrc file and run it as a command, profit.
